src/UI/Root/Components/Sidebar/Navigation/Navigation.test.tsx

```diff
@@ -171,24 +180,27 @@ test("GIVEN Navigation WHEN on 'Service Catalog' THEN 'Service Catalog' is highl
   expect(link).toHaveClass("active");
 });
 
-test("GIVEN Navigation WHEN Compilation Reports are pending THEN 'Compile Reports' Indication is visible", async () => {
-  const { component, apiHelper } = setup(
-    ["/lsm/catalog"],
-    TestServerStatus.withLsm,
-  );
+test("GIVEN Navigation WHEN Compilation Reports are not pending THEN 'Compile Reports' Indication does not exist", async () => {
+  const { component } = setup(["/lsm/catalog"], TestServerStatus.withLsm);
 
   render(component);
+  const Indication = screen.queryByLabelText("CompileReportsIndication");
 
-  expect(apiHelper.pendingRequests).toHaveLength(1);
-  expect(apiHelper.pendingRequests[0]).toEqual({
-    method: "HEAD",
-    url: "/api/v1/notify/env",
-  });
+  expect(Indication).toBeNull();
+});
 
-  await act(async () => {
-    await apiHelper.resolve(200);
+test("GIVEN Navigation WHEN Compilation Reports are pending THEN 'Compile Reports' Indication is visible", async () => {
+  jest.spyOn(queryModule, "useGetCompilerStatus").mockReturnValue({
```

Comment:
Why the spy instead of the msw implementation?

Reply:
because there is no trigger in that specific scenario, I opted for shortest approach